File: cpp/Stack_Queue/deque.cpp
#include <stdio.h>
#include <deque>

using namespace std;

int main(int argc, char const *argv[])
{
    /* Initialize the deque */
    deque<int> deque;

    /* Enqueue elements */
    deque.push_back(2); // Add to the tail
    deque.push_back(5);
    deque.push_back(4);
    deque.push_front(3); // Add to the head
    deque.push_front(1);

    /* Access elements */
    int front = deque.front(); // The first element
    int back = deque.back();   // The last element

    /* Dequeue elements */
    deque.pop_front(); // The first element dequeued
    deque.pop_back();  // The last element dequeued

    /* Get the length of the deque */
    int size = deque.size();

    /* Check if the deque is empty */
    bool empty = deque.empty();
    return 0;
}

File: java/deque/deque.java
import java.util.Deque;
import java.util.LinkedList;

public class deque {
  public static void main(String[] args) {
    /* Initialize the deque */
    Deque<Integer> deque = new LinkedList<>();

    /* Enqueue elements */
    deque.offerLast(2); // Add to the tail
    deque.offerLast(5);
    deque.offerLast(4);
    deque.offerFirst(3); // Add to the head
    deque.offerFirst(1);

    /* Access elements */
    int peekFirst = deque.peekFirst(); // The first element
    int peekLast = deque.peekLast(); // The last element

    /* Dequeue elements */
    int popFirst = deque.pollFirst(); // The first element dequeued
    int popLast = deque.pollLast(); // The last element dequeued

    /* Get the length of the deque */
    int size = deque.size();

    /* Check if the deque is empty */
    boolean isEmpty = deque.isEmpty();
  }
}
